storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -633,6 +657,10 @@ public String toString() {
                     ", txnIndex=" + txnIndex.file().getName() +
                     '}';
         }
+
+        public boolean hasTxnIndex() {
+            return txnIndex.file().length() == 0;
```

Comment:
is this check valid? should we have to invert it?

Reply:
my mistake, fixing it